crates/bitcoin-da/src/service.rs

```diff
@@ -308,15 +308,10 @@ impl BitcoinService {
         let rollup_name = self.rollup_name.clone();
         let da_private_key = self.da_private_key.expect("No private key set");
 
-        let blob = match da_data {
-            DaData::ZKProof(proof) => {
-                let blob = borsh::to_vec(&proof).expect("Should serialize");
-                compress_blob(&blob)
-            }
-            DaData::SequencerCommitment(commitment) => {
-                borsh::to_vec(&commitment).expect("Should serialize")
-            }
-        };
+        let mut blob = borsh::to_vec(da_data).expect("DaData must serialize");
```

Comment:
Thats a fix, it should work. However the actual fix should be where we call DaData::from_slice, because it’s not DaData anymore. It’s a SequencerCommitment | ForcedTransaction

Reply:
To separate ZKProof and SequencerCommitment, checking whether the tx sender is sequencer or prover is sufficient, and we can deserialize directly into the respective struct rather than DaData. But I do not know whether we will be able to discriminate the ForcedTransaction in a similar way.

Even if we can distinguish that one somehow as well, I am not sure about the benefit we have from that. IMO having some sort of variant discriminator in the blob might be useful.

Reply:
That's totally my bad.

Yes, I agree there should be a discriminator. But it shouldn't be of DaData.

This is how it works for DA now:
1. We send DaData tx
2. It is split in either TransactionKindLightClient(Complete, Chunks) or TransactionKindBatchProof(SequencerCommitment, "ForcedTransaction").
3. extract_sequencer_commitments scans for TransactionKindBatchProof only, filtering out zkproofs.
4. So we are left with SequencerCommitment and "ForcedTransaction". So I believe we should add TransactionKindBatchProof as a discriminator.

Reply:
Or we should wrap it into enum BatchProof { SequencerCommitment, ForcedTransaction }